dashboard/modules/state/state_head.py

```diff
@@ -166,6 +167,13 @@ def _options_from_req(self, req: aiohttp.web.Request) -> ListApiOptions:
             if req.query.get("limit") is not None
             else DEFAULT_LIMIT
         )
+
+        if limit > RAY_MAX_LIMIT_FROM_API_SERVER:
```

Comment:
I moved this from the `ListApiOptions` init because since we are now exposing this as an env var, the server and the client side could actually have different values, i.e. server sets the max limit to 10, but the client is still seeing 100 w/o setting the env explicitly. 

I don't think it makes sense for asking users to do the below if we are already exposing a limit flag, so I made this part of the API server config only (rather than a client config)
```
RAY_MAX_LIMIT_FROM_API_SERVER = 100 ray list --limit 100
```